### DISCUSSION TITLE
Add providers for full class hierarchy inheritance

### DISCUSSION THREAD

**Discussion by JordanLongstaff** [LemonAppDev/konsist#594] [Request a Feature]
### Tell us more.

The parent provider for class declarations only analyzes direct parents. Consider this use case:

```
class A
class B : A
class C : B
```

If I want to search for classes that have A as a parent class, I would expect the search to yield both B and C as results. However, C doesn't count since it's not a _direct_ subclass of A by its declaration.

I want a provider that will return _all_ subclasses, not just the direct ones.

**Comment by igorwojda** [comment]
Thx for reporting.
This has hight priority high in our backlog.

**Comment by igorwojda** [comment]
Coming in `0.14.0`

**Comment by nataliapeterwas** [comment]
A new version v0.14.0 has been released. It contains the fix.
See [v0.14.0 release](https://github.com/LemonAppDev/konsist/releases/tag/v0.14.0)
